Documentation/contributing/testing/e2e.rst

```diff
@@ -416,13 +416,74 @@ Example how to run ginkgo using ``dlv``:
 
 	dlv test . -- --ginkgo.focus="Runtime" -ginkgo.v=true --cilium.provision=false
 
+Running End-To-End Tests In Other Environments via kubeconfig
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Running End-To-End Tests In Other Environments
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The end-to-end tests can be run with an arbitrary kubeconfig file. Normally the
+CI will use the kubernetes created via vagrant but this can be overridden with
+``--cilium.kubeconfig``. When used, ginkgo will not start a VM nor compile
+cilium. It will also skip some setup tasks like labeling nodes for testing.
 
-If you want to run tests in a different VM, you can use ``--cilium.SSHConfig`` to
-provide the SSH configuration of the endpoint on which tests will be ran. The
-tests presume the following on the remote instance:
+This mode expects:
+
+- The current directory is ``cilium/test``
+
+- A test focus with ``--focus``. ``--focus="K8s*"`` selects all kubernetes tests.
+
+- Cilium images as full URLs specified with the ``--cilium.image`` and
+  ``--cilium.operator-image`` options, with matching ``CILIUM_IMAGE`` and
+  ``CILIUM_OPERATOR_IMAGE`` environment variables.
+
+- A working kubeconfig with the ``--cilium.kubeconfig`` option
+
+- A populated K8S_VERSION environment variable set to the version of the cluster
+
+- If appropriate, set the ``CNI_INTEGRATION`` environment variable set to one
+  of ``flannel``, ``eks``, ``microk8s`` or ``minikube``. This selects matching
+  configuration overrides for cilium.
+  Leaving this unset for non-matching integrations is also correct.
+  For k8s environments that invoke an authentication agent, such as EKS and
+  ``aws-iam-authenticator``, set ``--cilium.passCLIEnvironment=true``
+
+An example invocation is
+
+::
+
+  CNI_INTEGRATION=eks K8S_VERSION=1.13 CILIUM_IMAGE="quay.io/cilium/cilium:latest" CILIUM_OPERATOR_IMAGE="quay.io/cilium/operator:latest" ginkgo --focus="K8s*" -noColor -- -cilium.provision=false -cilium.kubeconfig=~/.kube/config -cilium.image="quay.io/cilium/cilium:latest" -cilium.operator-image="quay.io/cilium/operator:latest" -cilium.passCLIEnvironment=true)
```

Comment:
Any reason now to point to `K8S_VERSION=1.14` ? I think that is the default now on EKS clusters.

Reply:
@ungureanuvladvictor I actually just picked what `kubectl version` showed me on the cluster I was working with. I'll spin up another to see if it's changed (maybe it's regional? I was running in us-west-2).

Reply:
This is what I got from `kubectl version` running in `us-east-1a`:
```Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.8-eks-b8860f", GitCommit:"b8860f6c40640897e52c143f1b9f011a503d6e46", GitTreeState:"clean", BuildDate:"2019-11-25T00:55:38Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}```

Also I think you have a typo at the end of the cmd `)` -- looks not matched (same for line 476).

Reply:
oops! yes, the `)` are from my invocation (I have it in a subshell). Good catch.

I guess EKS is a random draw for versions :/
My us-west-2 output
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-04T04:48:55Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.12-eks-c500e1", GitCommit:"c500e11584c323151d6ab17526d1ed7461e45b0c", GitTreeState:"clean", BuildDate:"2019-10-22T03:11:52Z", GoVersion:"go1.11.13", Compiler:"gc", Platform:"linux/amd64"}
```

I'll point out the version a bit better... this is most certainly unexpected.

Reply:
Sounds good 👍

Reply:
@ungureanuvladvictor I added a little note. I'm not sure that it's obvious enough; I missed it and I was looking for it!
I switched the version to 1.14 for the EKS section, as many people will default to a us-east and 1.14 seems to be the version there. I left it as 1.13 in the more generic kubeconfig section because that is the default elsewhere in that doc. Do you think that's confusing?

Reply:
I think that is totally fine -- another thing we can do is paste a snippet like:
```
eksctl get cluster -r $region -p $profileName -n $clusterName -o json | jq -r '.[].Version'
```

in the docs and show people how they can get the K8S_VERSION via a one liner. 

I suspect a lot of folks use EKS via `eksctl`. I'm passing `-r` and `-p` since I'm operating in a non standard `eksctl` region.